Following iTero Gaming's "The Riot API Guide" playlist on YouTube to learn how to use Riot Game's API.

In [13]:
api_key = "RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650"

In [14]:
api_key

'RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'

In [15]:
api_url = "https://na1.api.riotgames.com/lol/summoner/v4/summoners/by-name/TeaNomad"

In [16]:
import requests

In [17]:
requests.get(api_url)

<Response [401]>

Get a 401 error because api_url does not have the API key attached, so when the request is made, there's nothing in the URL to say we have access to the information we want.
To fix that, need to combine api_url and api_key using the ? (argument). Now when making a request the API key is seen, and we're able to access the information.

In [18]:
api_url = api_url + '?api_key=' + api_key

In [19]:
resp = requests.get(api_url)

In [20]:
player_info = resp.json()

In [21]:
player_account_id = player_info['accountId']

In [22]:
player_account_id

'GHnLQSIE1GXtPfNXK7_gTTn8_Ly1Uym7qTVYmbWOw3bbiIv5LASkaTf2'

In [23]:
player_info

{'id': 'uTaSwiF1rpIshsGWRNCHK-j2OKToNm2gQqvy5WSIY20mVfzIyq7CqlO4bw',
 'accountId': 'GHnLQSIE1GXtPfNXK7_gTTn8_Ly1Uym7qTVYmbWOw3bbiIv5LASkaTf2',
 'puuid': 'AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w',
 'name': 'Tea Nomad',
 'profileIconId': 5089,
 'revisionDate': 1704757643000,
 'summonerLevel': 73}

In [24]:
api_url = 'https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w/ids?start=0&count=20'

In [25]:
api_url

'https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w/ids?start=0&count=20'

In [26]:
api_url = api_url + '&api_key=' + api_key

In [27]:
api_url

'https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w/ids?start=0&count=20&api_key=RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'

In [28]:
resp = requests.get(api_url)

In [29]:
matches = resp.json()

In [30]:
match = matches[0]
match

'NA1_4508715367'

In [31]:
api_url = 'https://americas.api.riotgames.com/lol/match/v5/matches/NA1_4508715367'
api_url

'https://americas.api.riotgames.com/lol/match/v5/matches/NA1_4508715367'

In [32]:
api_url = api_url + '?api_key=' + api_key
api_url

'https://americas.api.riotgames.com/lol/match/v5/matches/NA1_4508715367?api_key=RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'

In [33]:
resp = requests.get(api_url)
match_data = resp.json()
match_data.keys()

dict_keys(['metadata', 'info'])

In [34]:
match_data['metadata']

{'dataVersion': '2',
 'matchId': 'NA1_4508715367',
 'participants': ['iPfFbxEbDzT52UplVvAWpEy1uiT4UEw2KDhGrleLv6Qx0KTzL0Mv3c8E-jmlPguVZP8vH6YKW0aXtg',
  'AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w',
  'iintSefHBLL_5z3_dCOY0Yt41vNyVHKQWZuNzsEMkfvMlFjJpH_Kml57eNWUag3BLcxWxBwlwAMRCQ',
  'FHeET_9Qm1IgG5__Cmfug8P47vaBUxrsVg6IT1PdoIPJ8ys3Pj36P-NX_6wBIG0NWYzEgHWPdW26iw',
  '_fozelrA0BNk66cGM1BSfG-1T8MkVooPvQZOV2ONFLc42O8TmZuiWQrfYFAabU-n-UzYvt0yGHXTTg',
  'ZKsqyutqf2v8P-5pG-o4QTrf_4vGvbV9wjrJ-T0C27WDfGaDjC0uvc5tX1TFUbN4aogKEmJNzT4EFg',
  'f8_5MsVyQ5iXFAfoCZvgYkZIF6mzaufjO9wljCDMUwNdVNCVBHSgPor6Eo2KyxBNbUN9YqLrJRl-Lw',
  'Tl1nadJkvM9ZtNwAGlGswJJKxmp9M6EKVrNx5xZhZ-wmFy_NND4cVrA2zrRnGzgzjHXQZPZ_AaiTHQ',
  'k9J_xtjDP4m8qht7sUrZbE37P6lfe-oLIJbcgNk0PM43YWaLPiKUVY4V9YzqkZie-hYa6Ke8gpsLdQ',
  'JYcfEgiODYK47i1A-p3bbbXGuX1eZ2tEP6UJr28yVD-JTEdBg-EfinYIr1zYc8Haqyk6l9N9HLnCLQ']}

In [35]:
puuid = player_info['puuid']

In [36]:
part_index = match_data['metadata']['participants'].index(puuid)

In [37]:
kills = match_data['info']['participants'][part_index]['kills']
deaths = match_data['info']['participants'][part_index]['deaths']
assists = match_data['info']['participants'][part_index]['assists']
print((kills + deaths)/assists)

1.8


In [ ]:
api_url = 'https://americas.api.riotgames.com/lol/match/v5/matches/NA1_4508715367?api_key=RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'

In [68]:
import time

In [71]:
def get_match_data(region, match_id, api_key):
    api_url = (
        'https://' +
        region +
        '.api.riotgames.com/lol/match/v5/matches/' +
        match_id +
        '?api_key=' +
        api_key
    )
    while True:
        resp = requests.get(api_url)

        if resp.status_code == 429:
            print('Error: Rate limit exceeded. Please wait...')
            time.sleep(10)
            continue
        data = resp.json()
        return data

In [43]:
def did_win(puuid, match_data):
    match_data['metadata']['participants'].index(puuid)
    return match_data['info']['participants'][part_index]['win']

In [41]:
region = 'AMERICAS'
match_id = 'NA1_4508715367'
api_key = 'RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'
match_data = get_match_data(region, match_id, api_key)

{'metadata': {'dataVersion': '2',
  'matchId': 'NA1_4508715367',
  'participants': ['iPfFbxEbDzT52UplVvAWpEy1uiT4UEw2KDhGrleLv6Qx0KTzL0Mv3c8E-jmlPguVZP8vH6YKW0aXtg',
   'AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w',
   'iintSefHBLL_5z3_dCOY0Yt41vNyVHKQWZuNzsEMkfvMlFjJpH_Kml57eNWUag3BLcxWxBwlwAMRCQ',
   'FHeET_9Qm1IgG5__Cmfug8P47vaBUxrsVg6IT1PdoIPJ8ys3Pj36P-NX_6wBIG0NWYzEgHWPdW26iw',
   '_fozelrA0BNk66cGM1BSfG-1T8MkVooPvQZOV2ONFLc42O8TmZuiWQrfYFAabU-n-UzYvt0yGHXTTg',
   'ZKsqyutqf2v8P-5pG-o4QTrf_4vGvbV9wjrJ-T0C27WDfGaDjC0uvc5tX1TFUbN4aogKEmJNzT4EFg',
   'f8_5MsVyQ5iXFAfoCZvgYkZIF6mzaufjO9wljCDMUwNdVNCVBHSgPor6Eo2KyxBNbUN9YqLrJRl-Lw',
   'Tl1nadJkvM9ZtNwAGlGswJJKxmp9M6EKVrNx5xZhZ-wmFy_NND4cVrA2zrRnGzgzjHXQZPZ_AaiTHQ',
   'k9J_xtjDP4m8qht7sUrZbE37P6lfe-oLIJbcgNk0PM43YWaLPiKUVY4V9YzqkZie-hYa6Ke8gpsLdQ',
   'JYcfEgiODYK47i1A-p3bbbXGuX1eZ2tEP6UJr28yVD-JTEdBg-EfinYIr1zYc8Haqyk6l9N9HLnCLQ']},
 'info': {'gameCreation': 1670128369588,
  'gameDuration': 1585,
  

In [45]:
player_info['puuid']

'AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w'

In [46]:
print(did_win(puuid, match_data))

True


In [48]:
for match_id in matches:
    match_data = get_match_data(region, match_id, api_key)
    print(did_win(puuid, match_data))

True
True
False
True
False
True
False
True
False
False
True
False
True
True
False
False
True
True
True
True


In [49]:
api_url = 'https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/AcwemG5ZJeja0xX_yNXGKkKTUjlB12Ra_0fdQrOfOGlX7ouxQl81KeYbFXfFIENnR_BG-A2H4Arb_w/ids?start=0&count=100'

In [52]:
def get_matches(region, puuid, count, api_key):
    api_url = (
        'https://' +
        region +
        '.api.riotgames.com/lol/match/v5/matches/by-puuid/' +
        puuid +
        '/ids' +
        '?start=0&' +
        'count=' +
        str(count) + 
        '&api_key=' +
        api_key
    )

    resp = requests.get(api_url)
    return resp.json()

In [58]:
matches = get_matches(region, puuid, 100, api_key)

In [63]:
matches.extend(matches)
for match_id in matches:
    match_data = get_match_data(region, match_id, api_key)
    print(did_win(puuid, match_data))

True
True
False
True
False
True
False
True
False
False
True
False
True
True
False
False
True
True
True
True
False
False
True
False
True
True
True
False
True
False
True
False
False
False
True
True
True
False
True
True
False
False
True
True
True
True
False
True
False
False
True
True
True
True
False
False
False
False
False
True
False
True
True
True
True
False
False
True
False
True
True
True
True
False
False
True
False
False
False
False
True
False
True
False
True
False
False
True
False
False
True
True
False
False
True
False
False
True
True
False


KeyError: 'metadata'

In [64]:
api_url = 'https://americas.api.riotgames.com/lol/match/v5/matches/NA1_4508715367?api_key=RGAPI-18353aea-31f7-488d-9ef9-f777c3e16650'
resp = requests.get(api_url)
resp
resp.status_code == 429

True

Would get a <Response [429]> if the rate limit was exceeded.

In [72]:
def get_match_data(region, match_id, api_key):
    api_url = (
        'https://' +
        region +
        '.api.riotgames.com/lol/match/v5/matches/' +
        match_id +
        '?api_key=' +
        api_key
    )
    while True:
        resp = requests.get(api_url)

        if resp.status_code == 429:
            print('Error: Rate limit exceeded. Please wait...')
            time.sleep(10)
            continue
        data = resp.json()
        return data

Going through the matches twice because it was extended. Gave insight as to how bigger applications might work with the requests they make.

In [73]:
game_no = 1

for match_id in matches:
    print(game_no)
    print(match_id)
    match_data = get_match_data(region, match_id, api_key)
    print(did_win(puuid, match_data))
    print('')

    game_no += 1

1
NA1_4508715367
True

2
NA1_4508681071
True

3
NA1_4508646374
False

4
NA1_4508602692
True

5
NA1_4507054712
False

6
NA1_4507031100
True

7
NA1_4506286981
False

8
NA1_4506242262
True

9
NA1_4506200963
False

10
NA1_4506152173
False

11
NA1_4504805008
True

12
NA1_4504764233
False

13
NA1_4504728067
True

14
NA1_4504670399
True

15
NA1_4504632082
False

16
NA1_4504595168
False

17
NA1_4503930760
True

18
NA1_4503863636
True

19
NA1_4503822652
True

20
NA1_4503789687
True

21
NA1_4503773018
False

22
NA1_4503739356
False

23
NA1_4503726849
True

24
NA1_4503676921
False

25
NA1_4502937044
True

26
NA1_4502911958
True

27
NA1_4499729957
True

28
NA1_4499671474
False

29
NA1_4499649723
True

30
NA1_4499634562
False

31
NA1_4498878680
True

32
NA1_4498860205
False

33
NA1_4494782716
False

34
NA1_4494768093
False

35
NA1_4494678302
True

36
NA1_4494625960
True

37
NA1_4493624153
True

38
NA1_4493594951
False

39
NA1_4493539373
True

40
NA1_4493502357
True

41
NA1_4493463881
False

42
NA1_